include/mw/common/Lock.h

```diff
@@ -132,14 +133,14 @@ class Writer : virtual public Reader<T>
             OnEndWrite();
         }
 
-        void Commit()
+        void Commit(const std::unique_ptr<libmw::IDBBatch>& pBatch = nullptr)
```

Comment:
Traits::IBatchable shouldn't know anything about libmw::IDBBatch. You can actually just stop using Traits::IBatchable wherever you need to, since we don't really use the `Locked` wrapper anymore.